src/AutoRest.CSharp/Mgmt/Decorator/PropertyMatchDetection.cs

```diff
@@ -32,11 +37,43 @@ internal static bool IsEqual(List<PropertyInfo> parentProperties, List<ObjectTyp
 
             foreach (var childProperty in childProperties)
             {
-                if (!DoesPropertyExistInParent(childProperty, parentDict, propertiesInComparison))
+                if (!DoesPropertyExistInParent(sourceType, targetType, childProperty, parentDict, propertiesInComparison))
+                {
+                    if (allowExtra)
+                        continue;
+
                     return false;
+                }
+
+                matchCount++;
             }
 
-            return true;
+            return matchCount == parentProperties.Count;
```

Comment:
Ah I see - we still need to match all the parent properties, but it is okay if there are extra child properties. Still I think there could be a problem here because we are not ensuring that the match occurs for the InternalIncludes properties.

Reply:
not sure I follow the question, can you elaborate?

Reply:
I answered my own question 😄 
I didn't follow the algorithm initially.